utils/src/temp_data.rs

```diff
@@ -0,0 +1,99 @@
+use crate::buffer::Writable;
+
+pub struct TempData {
+    pub data: [u8; TempData::MAX_SIZE],
+    pub size: usize,
+}
+
+impl TempData {
+    pub const MAX_SIZE: usize = 160;
+
+    pub fn new() -> Self {
+        Self {
+            data: [0; TempData::MAX_SIZE],
+            size: 0,
+        }
+    }
+
+    #[inline]
+    pub fn reset(&mut self) {
+        self.data = [0; TempData::MAX_SIZE];
+        self.size = 0;
+    }
+
+    #[inline]
+    pub fn is_overflow(&self) -> bool {
+        self.size == Self::MAX_SIZE
+    }
+
+    #[inline]
+    pub fn write_byte(&mut self, byte: u8) {
+        if self.size == TempData::MAX_SIZE {
+            return;
```

Comment:
I changed it to `panic!()` since the `TempData` struct is only for testing, we checked the result here: https://github.com/alephium/ledger-alephium/blob/976928ef469ff936dc8ee7e8db61a5d600ce1f56/app/src/ledger_sdk_stub/swapping_buffer.rs#L142

and if it fails we will return an error.